File: backend/app/schemas/order.py
from pydantic import BaseModel
from datetime import datetime
from .product import Product


class OrderProduct(BaseModel):
    quantity: int
    total: float
    data: Product


class OrderBase(BaseModel):
    subtotal: float
    tax: float
    total: float
    products: list[OrderProduct]


class OrderCreate(OrderBase):
    pass


class Order(OrderBase):
    id: str
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/interfaces/order.ts
import { PaginationObj } from './pagination'
import { Product } from './product'

export interface OrderProduct {
	quantity: number
	total: number
	data: Product
}

export interface Order {
	id: string
	subtotal: number
	tax: number
	total: number
	products: OrderProduct[]
	created_at: string
}

export interface OrderQuery {
	search?: string
}

export interface PostData extends Omit<Order, 'id' | 'created_at'> {}

export interface PaginatedOrder extends PaginationObj {
	items: Order[]
}
